### ISSUE TITLE
Unable to edit objects

### ISSUE THREAD

**Issue by VantageKR** [NakedObjectsGroup/NakedObjectsFramework#173]
This is probably incorrect configuration on my part, but struggling to figure out what exactly changed to cause this...
The application can be browsed just fine, and pages can be refreshed, and the server API accessed to get the JSON responses, but when I try to edit an object, right after hitting the save button, the following error is returned by the client:

```
Client Error
Message: Failed to connect to server: http://[internal domain]:5000/objects/[type of object]/[id of object]
Code: ConnectionProblem
Description: The requested view of unsaved object details has expired.
Stack Trace : 
No stack trace available
   ```

I've replaced specifics with [] place holders. Note that when I paste that whole URL in a browser, the server API responds properly as per usual.

**Comment by VantageKR** [comment]
It's running on IIS 10. The error is not encountered when developing and running through IIS Express.

**Comment by richardpawson** [comment]
Have a look in Fiddler. It suggests the UI is calling Save and then is asking for the unsaved view to be re-loaded -  rather than simply displaying the new persisted object -  which shouldn't happen.  Is it possible this is down to accidental double-clicking or key-repeat, for example?  Is it the case for all browsers? Does it happen for other users?

**Comment by VantageKR** [comment]
Thank you very much Richard, for pointing me at Fiddler: a life saver once again!
First I tried a different browser, Edge, but had the same response.
Then I ran a Fiddler trace, and got the below:

```
HTTP/1.1 405 Method Not Allowed

Allow: GET, HEAD, OPTIONS, TRACE

Content-Type: text/html
```

..followed by HTML saying the same thing; HTTP verb not allowed.

So NOF was off the hook; it is an IIS issue! What happened is that I deployed to the same web server where I previously had to do configuration to allow the additional HTTP verbs required for the API to work, but this bit of config silently got overwritten.

Another thing was the red herring of me all the time thinking it was a NOF client issue, or config on the NOF client. I thought this due to the very quick error response and also since the client did not surface this bit of HTML that the API server was returning - maybe if the client cannot interpret the server API response, it could print that out in some way perhaps?  

After restoring the appropriate IIS configuration, all seems to be fine again, and I can breeth again...

This is the config part I had to re include in the web.config of the API:
    
```
  <system.webServer>
    <modules>
      <remove name="WebDAVModule" />
      <remove name="FormsAuthenticationModule" />
      <remove name="UrlRoutingModule-4.0" />
      <add name="UrlRoutingModule-4.0" type="System.Web.Routing.UrlRoutingModule" preCondition="" />
    </modules>
    <handlers>
      <remove name="WebDAV" />
      <remove name="ExtensionlessUrlHandler-Integrated-4.0" />
      <remove name="OPTIONSVerbHandler" />
      <remove name="TRACEVerbHandler" />
      <add name="ExtensionlessUrlHandler-Integrated-4.0" path="*." verb="*" type="System.Web.Handlers.TransferRequestHandler" preCondition="integratedMode,runtimeVersionv4.0" />
    </handlers>
  </system.webServer>
```

More on this general issue here:
https://docs.microsoft.com/en-us/aspnet/web-api/overview/testing-and-debugging/troubleshooting-http-405-errors-after-publishing-web-api-applications
